### ISSUE TITLE
PubSub 0.14.5 and streaming pull rpcs

### ISSUE THREAD

**Issue by ehacke** [GoogleCloudPlatform/google-cloud-node#2682]
#### Environment details

  - OS: Ubuntu
  - Node.js version: 8.7.0
  - npm version: 5.5.1
  - google-cloud-node version: 0.14.5
  - GKE version: 1.6.7

#### Steps to reproduce

1. Deploy many microservices with 0.13.2
2. No rate limiting
3. Update to 0.14.5
4. Lots of rate limiting

Is it possible that the 0.14.5 version of the pubsub client uses streaming pull rpcs?

I ask because since upgrading from 0.13.2 to 0.14.5 we've suddenly started hitting rate limits with our PubSub use, and it seems that the streamingpullrpcs quota is the one we're exceeding. Our use of the client is the same, we just upgraded the version and now we're being rate limited. Seems the default quota for streamingpullrpcs is 1000.

Also, after contacting our GCP rep about this, they seemed to be confused as to why we're using streamingpullrpcs because apparently it's a alpha/beta feature that is not ready for general production use?

Any thoughts? Any ideas why we'd suddenly see rate limiting just by bumping a client version?

**Comment by ehacke** [comment]
Not sure if this is related or not, but it also looks like after the update the Modify Ack Deadline operations for one of our subscriptions is extremely high: 
![modifyack](https://user-images.githubusercontent.com/4016786/31775862-ceef918c-b4b7-11e7-8adc-42a3332c7c96.png)

The above chart shows operations over the last hour.

**Comment by callmehiphop** [comment]
> Is it possible that the 0.14.5 version of the pubsub client uses streaming pull rpcs?

Yep! Per the PubSub teams request, we've implemented the StreamingPull RPC as of `0.14.0`.

> it also looks like after the update the Modify Ack Deadline operations for one of our subscriptions is extremely high

With `0.14.x` we've implemented a new feature we're calling *auto leasing*, where we constantly modify the ack deadline of messages that have yet to be acknowledged, preventing them from being redelivered. Right now it's not possible to opt out of, but configurations to allow that are coming soonish.
